pkg/crypto/certificatemanager/certificate_manager.go

```diff
@@ -150,19 +182,23 @@ func (m *manager) GetTLSContext(ctx context.Context, tlsCtx *api.TLSContext, ns
 	if ok {
 		private = string(privateBytes)
 	} else if tlsCtx.PrivateKey != "" {
-		return "", "", "", fmt.Errorf("Private Key %s not found in secret %s", privateName, name)
+		return "", "", "", false, fmt.Errorf("Private Key %s not found in secret %s", privateName, name)
 	}
 
 	if caBytes == nil && publicBytes == nil && privateBytes == nil {
-		return "", "", "", fmt.Errorf("TLS certificates not found in secret %s ", name)
+		return "", "", "", false, fmt.Errorf("TLS certificates not found in secret %s ", name)
 	}
 
-	return ca, public, private, nil
+	// TODO(youngnick): Follow up PR that will change this to a deprecation warning once we actually
+	// mark read-from-file as deprecated.
+	m.Logger.WithField("secret", name).Debug("Secret read from file")
+	return ca, public, private, fromFile, nil
 }
 
 // GetSecretString returns a secret string stored in a k8s secret
 func (m *manager) GetSecretString(ctx context.Context, secret *api.Secret, ns string) (string, error) {
-	name, secrets, err := m.GetSecrets(ctx, secret, ns)
+	// TODO: Update this and to handle read-from-file
+	name, secrets, _, err := m.GetSecrets(ctx, secret, ns)
```

Comment:
I actually updated this while you were reviewing, sorry.

The behavior if secret sync is disabled is now that secrets will be inlined as today. I'd prefer to have a way to replicate the current behavior in case there is something we've missed in the SDS-and-NPDS handling code in Envoy. I don't think it's likely there's a problem, but this lets users move back to the current behavior by changing a flag.